src/TcpPolyfill.js

```diff
@@ -56,14 +56,45 @@ export function Socket(options) {
 
     ws.onmessage = event => {
       const data = event.data;
+
+      // Handle incoming message as base64 if ws.protocol is set to base64
+      // or if it's not set at all and we are in a problematic React Native
+      // environment per:
+      // https://tools.ietf.org/html/rfc6455#section-11.3.4
+      // https://tools.ietf.org/html/rfc3864
+
+      const base64SetByDevice = (
```

Comment:
Maybe it's more apt to rename this `base64SetByServer`, since this is what the server ultimately agreed to be sending.